torchtitan/checkpoint.py

```diff
@@ -313,6 +315,19 @@ def save(self, curr_step: int, force: bool = False) -> None:
         else:
             dcp.save(self.states, checkpoint_id=checkpoint_id)
         self.reset()
+        if self.keep_latest_k > 0:
+            discovered_checkpoints = []
+            for filename in os.listdir(self.folder):
+                match = re.search(r"step-(\d+)", filename)
+                path = os.path.join(self.folder, filename)
+                discovered_checkpoints.append((int(match.group(1)), path))
+
+            discovered_checkpoints.sort()
+            to_delete = discovered_checkpoints[: -1 * self.keep_latest_k]
+
+            for _, path in to_delete:
+                logger.info(f"Deleting old checkpoint {path}")
+                shutil.rmtree(path, ignore_errors=True)
```

Comment:
nit: can we move this logic to a helper function to make `save()` simple to just call the helper?